File: backend/api-gateway/app/models/transit.py
from typing import Literal
from pydantic import BaseModel
from app.models.common import DataQuality, SourceItem


class TransitRealtimeRequest(BaseModel):
    session_id: str
    origin: str
    destination: str
    mode: Literal['subway', 'bus', 'either'] = 'either'


class TransitDeparture(BaseModel):
    route_id: str
    stop_name: str
    departure_time: str
    minutes_until_departure: int
    delay_seconds: int | None = None


class TransitRealtimeResponse(BaseModel):
    mode: Literal['subway', 'bus']
    origin_stop: str
    destination: str
    walking_to_stop_minutes: int
    waiting_minutes: int
    in_vehicle_minutes: int
    total_minutes: int
    recommended_leave_at: str
    estimated_arrival_at: str
    realtime_used: bool
    fallback_used: bool
    departures: list[TransitDeparture]
    data_quality: DataQuality
    source: list[SourceItem]

File: frontend/src/types/transit.ts
import type { DataQuality, SourceItem } from './api';

export type TransitRealtimeRequest = {
  session_id: string;
  origin: string;
  destination: string;
  mode: 'subway' | 'bus' | 'either';
};

export type TransitDeparture = {
  route_id: string;
  stop_name: string;
  departure_time: string;
  minutes_until_departure: number;
  delay_seconds?: number;
};

export type TransitRealtimeResponse = {
  mode: 'subway' | 'bus';
  origin_stop: string;
  destination: string;
  walking_to_stop_minutes: number;
  waiting_minutes: number;
  in_vehicle_minutes: number;
  total_minutes: number;
  recommended_leave_at: string;
  estimated_arrival_at: string;
  realtime_used: boolean;
  fallback_used: boolean;
  departures: TransitDeparture[];
  data_quality: DataQuality;
  source: SourceItem[];
};
